File: viewer-dotnet-winform-vb/CustomSearchInterface.vb
Public Class CustomSearchInterface

    Public Sub New()
        MyBase.New()
        InitializeComponent()
        ' In the form constructor do not use any Viewer methods (Open, Print etc.) that perform UI activities.
    End Sub
    Private Sub pdfViewer1_SearchDialogShowing(sender As Object, e As ceTe.DynamicPDF.Viewer.SearchDialogShowingEventArgs) Handles pdfViewer1.SearchDialogShowing
        e.Cancel = True
        searchText.Focus()
    End Sub
    Private Sub find_Click(sender As Object, e As EventArgs) Handles find.Click
        Dim matchOptions As ceTe.DynamicPDF.Viewer.MatchOptions
        If (Not String.IsNullOrEmpty(searchText.Text)) Then
            matchOptions = ceTe.DynamicPDF.Viewer.MatchOptions.All
            If ((matchWholeWord.Checked = True) And (matchCase.Checked = True)) Then
                matchOptions = ceTe.DynamicPDF.Viewer.MatchOptions.All
            ElseIf ((matchWholeWord.Checked = True) And (matchCase.Checked = False)) Then
                matchOptions = ceTe.DynamicPDF.Viewer.MatchOptions.WholeWordOnly
            ElseIf ((matchWholeWord.Checked = False) And (matchCase.Checked = True)) Then
                matchOptions = ceTe.DynamicPDF.Viewer.MatchOptions.CaseSensitive
            Else
                matchOptions = ceTe.DynamicPDF.Viewer.MatchOptions.None
            End If
            pdfViewer1.SearchForward(searchText.Text, matchOptions)
        End If
    End Sub

    Private Sub pdfViewer1_DocumentDisplayed(sender As Object, e As ceTe.DynamicPDF.Viewer.DocumentDisplayedEventArgs) Handles pdfViewer1.DocumentDisplayed
        groupBox1.Enabled = True
        title.Text = "Type a search term in the 'Find' text box and click on 'Find Next'."
    End Sub
End Class

File: viewer-dotnet-winform-cs/CustomSearchInterface.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace viewer_dotnet_winforms_cs
{
    public partial class CustomSearchInterface : Form
    {
        public CustomSearchInterface()
        {
            InitializeComponent();
        }
        private void pdfViewer1_SearchDialogShowing(object sender, ceTe.DynamicPDF.Viewer.SearchDialogShowingEventArgs e)
        {
            e.Cancel = true;
            searchText.Focus();
        }

        private void button3_Click(object sender, EventArgs e)
        {
            if (!String.IsNullOrEmpty(searchText.Text))
            {
                ceTe.DynamicPDF.Viewer.MatchOptions matchOptions = ceTe.DynamicPDF.Viewer.MatchOptions.All; ;
                if ((matchWholeWord.Checked == true) && (matchCase.Checked == true))
                {
                    matchOptions = ceTe.DynamicPDF.Viewer.MatchOptions.All;
                }
                else if ((matchWholeWord.Checked == true) && (matchCase.Checked == false))
                {
                    matchOptions = ceTe.DynamicPDF.Viewer.MatchOptions.WholeWordOnly;
                }
                else if ((matchWholeWord.Checked == false) && (matchCase.Checked == true))
                {
                    matchOptions = ceTe.DynamicPDF.Viewer.MatchOptions.CaseSensitive;
                }
                else
                {
                    matchOptions = ceTe.DynamicPDF.Viewer.MatchOptions.None;
                }

                pdfViewer1.SearchForward(searchText.Text, matchOptions);
            }
        }

        private void pdfViewer1_DocumentDisplayed(object sender, ceTe.DynamicPDF.Viewer.DocumentDisplayedEventArgs e)
        {
            groupBox1.Enabled = true;
            title.Text = "Type a search term in the 'Find' text box and click on 'Find Next'.";
        }
    }
}
